hazelcast/src/main/java/com/hazelcast/cache/impl/operation/AbstractCacheOperation.java

```diff
@@ -62,8 +67,11 @@ public String getServiceName() {
     @Override
     public final void beforeRun()
             throws Exception {
-        ICacheService service = getService();
-        cache = (CacheRecordStore) service.getOrCreateRecordStore(name, getPartitionId());
+        cacheService = getService();
+        if (cacheService.wanReplicationEnabled(name)) {
```

Comment:
we are adding this check to all cache operations, is it costly ?
can we add this check to `CacheRecordStore` as serkan suggessted ?